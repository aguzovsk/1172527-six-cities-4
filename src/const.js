const cityNames = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const rentTypes = [`apartment`, `room`, `house`, `hotel`];

const cityCoordinates = new Map();
cityCoordinates.set(`Amsterdam`, {
  latitude: 52.37454,
  longitude: 4.897976,
});
cityCoordinates.set(`Brussels`, {
  latitude: 50.846557,
  longitude: 4.351697,
});
cityCoordinates.set(`Cologne`, {
  latitude: 50.938361,
  longitude: 6.959974,
});
cityCoordinates.set(`Dusseldorf`, {
  latitude: 51.225402,
  longitude: 6.776314,
});
cityCoordinates.set(`Hamburg`, {
  latitude: 53.550341,
  longitude: 10.000654,
});
cityCoordinates.set(`Paris`, {
  latitude: 48.85661,
  longitude: 2.351499,
});

const cities = cityNames.map((name) => ({
  name,
  location: Object.assign({zoom: 12}, cityCoordinates.get(name))
})).reduce((map, cityObj) => {
  map.set(cityObj.name, cityObj);
  return map;
}, new Map());

export const SortTypes = {
  DEFAULT: `popular`,
  TO_HIGH: `to-high`,
  TO_LOW: `to-low`,
  TOP: `top-rated`
};

export const UNSELECTED_OFFER = null;
export const UNSELECTED_ITEM = undefined;
export const mapObjectCreator = () => new Map();

// const imgPath = `img`;
// const imageNames = [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-01.jpg`];

export {cityCoordinates, cities};
export {cityNames, rentTypes};
// export {imgPath, imageNames};
