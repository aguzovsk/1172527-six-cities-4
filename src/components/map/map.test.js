import React from 'react';
import renderer from 'react-test-renderer';
import Map from './map.jsx';
import offersAmsterdam from '../../test-mock/offers-amsterdam.js';
import {cities} from '../../const.js';

it(`Map snapshot test on Amsterdam`, () => {
  const tree = renderer.create(
      (<Map city={cities.get(`Amsterdam`)} offers={offersAmsterdam} />),
      {
        createNodeMock() {
          return document.createElement(`div`);
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
