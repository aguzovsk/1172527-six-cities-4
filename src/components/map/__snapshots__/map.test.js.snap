// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map snapshot test on Amsterdam 1`] = `
<div
  className="map undefined"
  id="map"
  style={
    Object {
      "width": "100%",
    }
  }
/>
`;
